pkg/docker/config/config_test.go

```diff
@@ -566,3 +624,54 @@ func TestGetAllCredentials(t *testing.T) {
 	}
 
 }
+
+func TestAuthKeysForRef(t *testing.T) {
+	for _, tc := range []struct {
+		name, ref string
+		expected  []string
+	}{
+		{
+			name: "image without tag",
+			ref:  "quay.io/image",
+			expected: []string{
+				"quay.io/image",
+				"quay.io",
+			},
+		},
+		{
+			name: "image with tag",
+			ref:  "quay.io/image:latest",
+			expected: []string{
+				"quay.io/image",
+				"quay.io",
+			},
+		},
+		{
+			name: "image single path tag",
+			ref:  "quay.io/user/image:latest",
+			expected: []string{
+				"quay.io/user/image",
+				"quay.io/user",
+				"quay.io",
+			},
+		},
+		{
+			name: "image with nested path",
+			ref:  "quay.io/a/b/c/d/image:latest",
+			expected: []string{
+				"quay.io/a/b/c/d/image",
+				"quay.io/a/b/c/d",
+				"quay.io/a/b/c",
+				"quay.io/a/b",
+				"quay.io/a",
+				"quay.io",
+			},
+		},
+	} {
+		ref, err := reference.ParseNamed(tc.ref)
+		require.Nil(t, err, tc.name)
```

Comment:
```suggestion
		require.NoError(t, err, tc.name)
```

Reply:
(`require.NoError` aborts the test, unlike `assert.NoError` continuing, and doing something unexpected with invalid `ref`. OTOH if the test fails, that’s quite enough and worrying about _how_ it fails can get unproductive.)